backend/src/main/java/ai/verta/modeldb/project/FutureProjectDAO.java

```diff
@@ -1357,4 +1359,85 @@ public FutureProjectDAO(
             },
             executor);
   }
+
+  public InternalFuture<Void> setProjectShortName(SetProjectShortName request) {
+    // Request Parameter Validation
+    InternalFuture<Void> validateParamFuture =
+        InternalFuture.runAsync(
+            () -> {
+              if (request.getId().isEmpty() && request.getShortName().isEmpty()) {
+                throw new InvalidArgumentException(
+                    "Project ID and Project shortName not found in SetProjectShortName request");
+              } else if (request.getId().isEmpty()) {
```

Comment:
This first check seems redundant. 

If we want to do this kind of response we should switch from immediately throwing an exception to a model where we have a validator that collects all violations and then throws a single exception listing all violations.

```suggestion
              if (request.getId().isEmpty()) {
```

Reply:
Removed redundant validation check